train/train.py

```diff
@@ -0,0 +1,65 @@
+import torch
+
+from utils import Config
+from torch.optim import AdamW
+from torch.optim.lr_scheduler import CosineAnnealingLR
+from tqdm import tqdm
+from torch.nn.functional import l1_loss
+from train.eval import eval_model
+from model import AOSRestoration
+from torch.utils.tensorboard import SummaryWriter
+
+import os
+import time
+
+
+def train(config: Config):
+    best_eval_psnr = float('-inf')
+    best_eval_ssim = float('-inf')
+    n_milestone = 0
+
+    writer = SummaryWriter(config.result_path)
+    model = AOSRestoration.get_model_from_config(config)
+    optimizer = AdamW(model.parameters(), lr=config.lr, weight_decay=1e-4)
+    lr_scheduler = CosineAnnealingLR(optimizer, T_max=config.num_iter, eta_min=1e-6)
+
+    progress_bar = tqdm(range(1, config.num_iter + 1), initial=1, dynamic_ncols=True)
+
+    for n_iter in progress_bar:
+        idx_iter = n_iter - 1
+        if n_iter == 1 or idx_iter in config.milestones:
+            end_iter = config.milestones[n_milestone] if n_milestone < len(config.milestones) else config.num_iter
+            start_iter = config.milestones[n_milestone - 1] if n_milestone > 0 else 0
+            length = config.batch_size[n_milestone] * (end_iter - start_iter)
+            train_dataset = None  # TODO
+            train_loader = None  # TODO
```

Comment:
todo after #17 is merged

Reply:
Please use:

train_loader, test_loader = data.get_aos_loaders(train_ratio=0.9, train_batch_size = 32, test_batch_size = 32)

or

# choose data sets to include in processing
G_DATASETS = {1: True,
              2: True,
              3: False}

# choose data set locations
G_DATASETS_LOCATION = {1: "../dataset/Part1",
                       2: "../dataset/Part2",
                       3: "../dataset/Part3"}

G_DATASET_LOCATION = [G_DATASETS_LOCATION[dataset_id] for dataset_id, include_dataset in G_DATASETS.items() if include_dataset]

dataset = data.AOSDataset(G_DATASET_LOCATION,
                          relative_path=True,
                          maximum_datasize=None)
                          
train_loader, test_loader = data.AOSDataloader(dataset, train_ratio=0.9)